Allow ix_create_error/ix_update_error to return successes also

wolfsage opened the pull request:
Perhaps trying to create a duplicate row should "succeed" and just return
the existing row in the database. Now, ix_create_error can do that

rjbs commented on t/lib/Bakesale/Schema/Result/User.pm:
```diff
@@ -32,24 +32,48 @@ sub ix_default_properties ($self, $ctx) {
 }
 
-sub ix_create_error ($self, $ctx, $error) {
+sub ix_create_error ($self, $ctx, $rec, $error) {
```
It may be that in the future we need to pass both the record we tried to create _and_ the record that the user asked us to create.  (The difference here is at least defaults.)  Maybe `$rec` should (a) some after `$error` and (b) be an entry in a hashref of arguments.

`$rclass->ix_create_error($ctx, $error, { rec => $rec })```

?

wolfsage replied:
Done!

rjbs commented on lib/Ix/DBIC/ResultSet.pm:
```diff
@@ -411,10 +411,16 @@ sub ix_create ($self, $ctx, $to_create) {
       my $exception = $@;
 
-      my $error;
+      my ($error, $res);
```
As discussed on Slack, I think that it should be the above eval (now to become try/catch) that should return these two, rather than doing it in the no-$row branch.  The way done here, you've missed logging the creation id, and other row-reporting logic.

rjbs replied:
I'm pretty sure that convention for this kind of return is always `($result, $error)`, not `($error, $result)`.  Even outside of Go.  :-)  Please follow suit.

wolfsage replied:
Right, got it

rjbs commented:
:+1: other than the notes!

wolfsage committed:
Allow ix_create_error/ix_update_error to return successes also

Perhaps trying to create a duplicate row should "succeed" and just return
the existing row in the database. Now, ix_create_error can do that.

rjbs commented:
Looks good!

wolfsage commented:
Thanks! *tries the fancy rebase button on github*
